### ISSUE TITLE
TreeViewItem doesn't always handle IsExpanded binding properly

### ISSUE THREAD

**Issue by arcadiogarcia** [microsoft/microsoft-ui-xaml#2112]
I am trying to expand/collapse values in a TreeView procedurally by binding the IsExpanded property in each TreeViewNode. This initially works, but there seems to be  a bug that break this behavior.

**Steps to reproduce the bug**
1. Clone and run this project https://github.com/arcadiogarcia/ListViewBugRepro
2. Click Refresh, tree gets populated
3. Click Expand, tree expands properly
4. Click Refresh, tree gets populated again
5. Click Expand, tree fails to expand properly. Now it can only expand the first level, or the last level that was closed automatically.

**Expected behavior**
The values of IsExpanded should still be honored after reassigning the value of ItemsSource. It seems that expansion only works correctly for the first tree that is assigned, and fails for the subsequent.

**Screenshots**
Tree after step 3:
![image](https://user-images.githubusercontent.com/9416356/76580167-3542f100-648c-11ea-9c42-de7e130427c7.png)
Tree after step 5, note that IsExpanded == true but the node is not expanded:
![image](https://user-images.githubusercontent.com/9416356/76580178-3aa03b80-648c-11ea-8f44-f71d271b5b7d.png)

**Version Info**
Running 18363

NuGet package version: 
Can repro with both the inbox controls and Microsoft.UI.Xaml 2.3.200213001 

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| November 2019 Update (18363)  | Yes   |
| May 2019 Update (18362)            |  |
| October 2018 Update (17763)        |  |
| April 2018 Update (17134)          |   |
| Fall Creators Update (16299)       |  |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 | Yes |
| Mobile                   |   |
| Xbox                      |   |
| Surface Hub          |     |
| IoT                         |     |

**Comment by arcadiogarcia** [comment]
Ok I just saw that this is a duplicate of #1790 , I didn't find it before because it was closed. When should i expect this to make it to a stable build?

**Comment by ranjeshj** [comment]
@arcadiogarcia the latest pre-release build should already have this fix. Can you use that ? The next stable release build is planned by the end of the month. Thanks.

**Comment by arcadiogarcia** [comment]
@ranjeshj I tried updating to version 2.3.200213001 but I can still repro the issue, here is my example: https://github.com/arcadiogarcia/ListViewBugRepro

**Comment by ranjeshj** [comment]
@kaiguo can you take a quick look please ? I expected this to be fixed with your change.

**Comment by michael-hawker** [comment]
@arcadiogarcia the original issue finally got a post from the bot saying it should be in release `v2.4.0-prerelease.200322001`

**Comment by michael-hawker** [comment]
I've tested this out now and this appears to work, though when I rebind the collection, I do see that it does 'blink' between showing the root and having it expanded.

**Comment by HLCaptain** [comment]
I have the same problem or a little bit different with my app. https://github.com/HLCaptain/Hypr
I use WinUI 2.61. When I have no project open, I bound some dummy data to my TreeView.
Related files:
- ViewModels/Editor/EditorTreeViewModel.cs
- Views/CustomControls/EditorTreeControl.xaml
- Views/CustomControls/EditorTreeControl.xaml.cs

What I did:
Opened node "No project" and "Child1"
![image](https://user-images.githubusercontent.com/22623259/128030598-a523df17-6850-48c0-adc7-0c4f9bc6126a.png)
- Closed "No project"
![image](https://user-images.githubusercontent.com/22623259/128030739-02081e09-4c2e-46c0-83e2-382d36e16d32.png)
Unexpected: "Child1" should be open.
- Opened "Child1" and closed "No project", then opened "No project" again.
Unexpected: "Child1" should be open. But it looks like, the "No project" node is some kind of bug loop, because from now on, "Child1"'s expansion toggles as I open and close "No project".

It gets worse with more data down the road. Children nodes' expansion toggles unexpectadly and interrupts workflow.
Bug appeared when I ported properly working solution from code-behind to MVVM (to ViewModel) and bound the data.

OS Build: 19043.1110
